client/extensions/zoninator/state/zones/reducer.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer } from 'state/utils';
+import {
+	ZONINATOR_FETCH_ERROR,
+	ZONINATOR_FETCH_ZONES,
+	ZONINATOR_UPDATE_ZONES
+} from '../action-types';
+
+export const fetching = createReducer( {}, {
+	[ ZONINATOR_FETCH_ZONES ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: true } ),
+	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+	[ ZONINATOR_FETCH_ERROR ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+} );
+
+export const zones = createReducer( {}, {
+	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId, data } ) => ( { ...state, [ siteId ]: data } ),
+} );
+
+export default combineReducers( {
+	fetching,
+	zones,
```

Comment:
Maybe better rename to `items` as we usually do -- to avoid name collisions with https://github.com/Automattic/wp-calypso/pull/16740/files#diff-77581f2fa47ffe29bb5b6a170f673e03R7.